src/Raven.Studio/typescript/components/pages/resources/manageServer/storageReport/partials/StorageReportTable.tsx

```diff
@@ -0,0 +1,101 @@
+import { hasChildren, StorageReportItem } from "components/pages/resources/manageServer/storageReport/partials/common";
+import genUtils from "common/generalUtils";
+import React, { useId } from "react";
+import { withPreventDefault } from "components/utils/common";
+import { UncontrolledTooltip } from "reactstrap";
+import useUniqueId from "hooks/useUniqueId";
+
+interface StorageReportTableProps {
+    node: StorageReportItem;
+    root: StorageReportItem;
+    onNodeSelected: (node: StorageReportItem) => void;
+}
+
+export function StorageReportTable(props: StorageReportTableProps) {
+    const { node, root, onNodeSelected } = props;
+    const tooltipId = useUniqueId("tooltip");
+
+    if (!node || !node.internalChildren) {
+        return null;
+    }
+
+    const showPagesColumn = node.internalChildren.some((x) => x.type === "tree");
+    const showEntriesColumn = node.internalChildren.some((x) => x.type === "table" || x.type === "tree");
+    const showTempFiles = node === root;
+
+    return (
+        <table className="table table-condensed table-striped on-base-background">
+            <thead>
+                <tr>
+                    <th>Type</th>
+                    <th className="column-min-width">Name</th>
+                    {showPagesColumn && <th key="pages-column"># Pages</th>}
+                    {showEntriesColumn && <th key="entries">Entries</th>}
+                    <th>
+                        {showTempFiles && <>Total</>}
+                        Size (&sum; <FormattedSize node={node} header />)
+                    </th>
+                    <th>% Total</th>
+                </tr>
+            </thead>
+            <tbody>
+                {node.internalChildren.map((item, index) => (
+                    <tr key={index}>
+                        <td>{_.upperFirst(item.type)}</td>
```

Comment:
Let's create replacement function for lodash. For now place it in `typeUtils.ts`. You can return `Capitalize<T>` for typseafty